house-cats/huyan
# feat: submit RsvpEditor form fields to backend

# Context

- This PR sets up the RSVP Editor form to submit an updated RSVP to the backend.
- ESLint flags a false positive for accessibility, complaining that the `<label>` tag should be associated with an `<input>` control. Formik's `<Field>` isn't recognized as a control, but it'll render an `<input>`, so I've disabled that lint for those lines.

# Testing

**Make sure to set up your `.env` file!** It should be in the `web/` directory, and follow the format of the `.env.example` file in there.

**The Deploy Preview will not work!** I have not yet set up `.env` files for it, so RSVP Lookup will fail.

1. Start the website up: `cd web/ && npm start`
2. Open the RSVP page in your browser, which should show the RSVP Lookup: http://localhost:8080/rsvp
3. Fill out the form as follows (it's based on the one user who with RSVP data in the Deta Base instance) — inputs are _**not** case sensitive_:
    - **First name:** furst
    - **Last name:** lahst
4. Click the **Submit** button.
5. The page will transition to the RSVP Editor with the **First name** and **Last name** fields prepopulated.
6. Update any of the fields and then click **Submit**.
7. You should see a small message above the **Submit** button that reads **Got the update, thanks!**. (This is a placeholder; we'll eventually just transition the user to a confirmation page restating what they entered, based on what the backend returns.)
8. Reload the page, and re-enter the same **First name** and **Last name** as earlier on the RSVP Lookup.
9. The RSVP Editor should now show the same exact data that you submitted earlier.
